### PR TITLE
Added Strict mode support

### PR DESCRIPTION
**Pull request by mohitsoni**
Summary of changes:
* Updated libmesos binary
* Installs injected certs into JRE
* Adds Strict mode support for:
  * cassandra
  * elastic
  * hdfs
  * kafka
  * hello-world
  * template
* Currently, strict mode is only supported with V0 (non-http) Mesos API. Once Mesos adds support for auth-n-auth for V1 (http) Mesos API, the frameworks can just switch to that via `MESOS_API_VERSION` variable.
* Also, refactors Kafka so that it does that right thing w.r.t. registration. This was required to ensure that FrameworkInfo was created in a consistent way like other frameworks.

### CODE REVIEW

File: frameworks/hdfs/universe/marathon.json.mustache [hunk 1]
```diff
@@ -40,6 +46,13 @@
     "LIBMESOS_URI": "{{resource.assets.uris.libmesos-bundle-tar-gz}}",
     "HDFS_URI": "{{resource.assets.uris.hdfs-tar-gz}}",
     "BOOTSTRAP_URI": "{{resource.assets.uris.bootstrap-zip}}",
+
+    {{#service.secret_name}}
+    "DCOS_SERVICE_ACCOUNT_CREDENTIAL": { "secret": "serviceCredential" },
+    "MESOS_MODULES": "{\"libraries\": [{\"file\": \"libdcos_security.so\", \"modules\": [{\"name\": \"com_mesosphere_dcos_ClassicRPCAuthenticatee\"}]}]}",
```
Comment (mrbrowning) [inline frameworks/hdfs/universe/marathon.json.mustache]: This should be unnecessary with the bundle that provides `modules.json` out of the box, right? Or is there something preventing us from using that bundle here?

File: frameworks/kafka/src/main/dist/svc.yml [hunk 2]
```diff
@@ -1,8 +1,6 @@
 name: {{FRAMEWORK_NAME}}
 scheduler:
-  principal: {{FRAMEWORK_PRINCIPLE}}
-  zookeeper: {{MESOS_ZOOKEEPER_URI}}
-  api-port: {{PORT_API}}
+  principal: {{FRAMEWORK_PRINCIPAL}}
```
Comment (mrbrowning) [inline frameworks/kafka/src/main/dist/svc.yml]: Does this one need a user too?

File: frameworks/hdfs/universe/marathon.json.mustache (hunk 1 above)
Comment (mohitsoni) [inline frameworks/hdfs/universe/marathon.json.mustache]: As per an offline conversation with @karya0, this is something that we need to include now.

File: frameworks/kafka/src/main/dist/svc.yml (hunk 2 above)
Comment (mohitsoni) [inline frameworks/kafka/src/main/dist/svc.yml]: Good catch. Let me update the PR.

File: frameworks/hdfs/universe/marathon.json.mustache (hunk 1 above)
Comment (mrbrowning) [inline frameworks/hdfs/universe/marathon.json.mustache]: Okay. @karya0 , maybe this is a good place to ask: why is this necessary? I thought modules were linked on agent launch, so how is this value being used? And will schedulers running in strict mode always have to care about what seems like a Mesos implementation detail?

**Comment by loren** [comment]
Juts curious, what sets elastic apart w/r/t strict mode? Is this one more way that it's a special snowflake?

Also, what about the cassandra FW?

File: frameworks/hdfs/universe/marathon.json.mustache (hunk 1 above)
Comment (gabrielhartmann) [inline frameworks/hdfs/universe/marathon.json.mustache]: We should address this ASAP.  @mohitsoni: Can you ping @karya0 out of band please?

**Comment by mohitsoni** [comment]
@loren Elastic needs a 1.9 cluster, and at the time of sending this PR there was a bug in 1.9-dev strict mode. And, because of that I was unable to verify my changes, and hence decided to not include elastic in this PR (until the bug gets fixed).

As far as the strict-mode support changes are concerned, elastic is not a snowflake.

File: frameworks/hdfs/universe/marathon.json.mustache (hunk 1 above)
Comment (karya0) [inline frameworks/hdfs/universe/marathon.json.mustache]: The `MESOS_MODULES` env requires an absolute path of the form `file:///` and for those frameworks that untar the bundle in their MESOS_SANDBOX directory, we can't predict the absolute path.

For framework that untar the bundle at a known location, i.e., `/opt/libmesos-bundle`, we can include a `mesos_scheduler_authenticatee_module.json` file in the bundle and let the app point to it.

Comment (gabrielhartmann) [inline frameworks/hdfs/universe/marathon.json.mustache]: @karya0:  Why is the absolute path required?  Can support for relative path be added?

File: frameworks/kafka/src/main/dist/svc.yml (hunk 2 above)
Comment (mohitsoni) [inline frameworks/kafka/src/main/dist/svc.yml]: Done!

**Comment by mohitsoni** [comment]
@loren Added strict mode support for Cassandra.

**Comment by mohitsoni** [comment]
Added strict mode support for elastic.

File: frameworks/hdfs/universe/marathon.json.mustache (hunk 1 above)
Comment (karya0) [inline frameworks/hdfs/universe/marathon.json.mustache]: @gabrielhartmann: It's because Mesos doesn't support it. We can try to get a patch into Mesos. Currently, Mesos just hands over the string to the 'open' call. What we need is to change this behavior to resolve the env vars before passing it to open. I'll lookup open semantics around env var resolution and write back with an update on the possibilities.

Comment (nickbp) [inline frameworks/hdfs/universe/marathon.json.mustache]: Out of curiosity, what does passing file:///dev/urandom do?

Comment (karya0) [inline frameworks/hdfs/universe/marathon.json.mustache]: @nickbp: The way Mesos stout processes the filepaths, it will first try to read the *entire* file contents into a buffer (8KB at a time) and then will parse it to JSON parser and will follow it by a JSON->Proto conversion. As you can guess, the reading will never finish up and at some point, you'll run out of memory+swap and the process will get killed.

**Comment by mohitsoni** [comment]
retest this please

File: frameworks/hdfs/universe/marathon.json.mustache (hunk 1 above)
Comment (nickbp) [inline frameworks/hdfs/universe/marathon.json.mustache]: I assume 'the process' in this case is the agent? When it restarts will it just try to do the same thing again?

Comment (karya0) [inline frameworks/hdfs/universe/marathon.json.mustache]: It would be whatever process was trying to load the module manifest. It could be the master, agent, or the scheduler.
